File: python/app/dynamodb_example.py
import boto3
from dotenv import load_dotenv
from mypy_boto3_dynamodb.client import DynamoDBClient

from app.env import (
    endpoint,
    aws_access_key_id,
    dynamodb_user_table,
    aws_secret_access_key,
)

load_dotenv()


def create_dynamodb_client() -> DynamoDBClient:
    boto3.client(
        "dynamodb",
        endpoint_url=endpoint,
        aws_access_key_id=aws_access_key_id,
        aws_secret_access_key=aws_secret_access_key,
    )


def create_item(dynamodb: DynamoDBClient, item):
    dynamodb.put_item(TableName=dynamodb_user_table, Item=item)

File: node/src/gateways/dynamodbExample.ts
import AWS from 'aws-sdk';
import dotenv from 'dotenv';

dotenv.config();

const dynamoDBConfig = {
  endpoint: process.env.ENDPOINT,
  region: process.env.REGION,
};

export const createDynamoDBClient = () => {
  return new AWS.DynamoDB.DocumentClient(dynamoDBConfig);
};

export const createItem = async (db: AWS.DynamoDB, tableName: string, item: any) => {
  const params = {
    TableName: tableName,
    Item: item,
  };

  return db.putItem(params).promise();
};

export const readItem = async (db: AWS.DynamoDB, tableName: string, key: any) => {
  const params = {
    TableName: tableName,
    Key: key,
  };

  return db.getItem(params).promise().then((data) => data.Item);
};

export const updateItem = async (db: AWS.DynamoDB, tableName: string, key: any, updateExpression: string, expressionAttributeValues: any) => {
  const params = {
    TableName: tableName,
    Key: key,
    UpdateExpression: updateExpression,
    ExpressionAttributeValues: expressionAttributeValues,
    ReturnValues: 'ALL_NEW',
  };

  return db.updateItem(params).promise().then((data) => data.Attributes);
};

export const deleteItem = async (db: AWS.DynamoDB, tableName: string, key: any) => {
  const params = {
    TableName: tableName,
    Key: key,
    ReturnValues: 'ALL_OLD',
  };

  return db.deleteItem(params).promise().then((data) => data.Attributes);
};
